mrmustard/lab_dev/states/base.py

```diff
@@ -497,6 +499,28 @@ def _repr_html_(self):  # pragma: no cover
         template = Template(filename=os.path.dirname(__file__) + "/assets/states.txt")
         display(HTML(template.render(state=self)))
 
+    def _getitem_builtin_state(self, modes: set[int]):
+        r"""
+        A convenience function to slice built-in states.
+
+        Built-in states come with a parameter set. To slice them, we simply slice the parameter
+        set, and then used the sliced parameter set to re-initialize them.
+
+        This approach avoids computing the representation, which may be expensive. Additionally,
+        it allows returning trainable states.
+        """
+        # slice the parameter set
+        items = [i for i, m in enumerate(self.modes) if m in modes]
+        kwargs = {}
+        for name, param in self._parameter_set[items].all_parameters.items():
+            kwargs[name] = param.value
+            if isinstance(param, Variable):
+                kwargs[name + "_trainable"] = True
+                kwargs[name + "_bounds"] = param.bounds
+
+        # use `mro` to return the correct state
+        return self.__class__.mro()[0](modes, **kwargs)
```

Comment:
Can you remind me why this is necessary? When is `self.__class__.mro()[0]` not equal to `self.__class__`?

Reply:
No you're right, we can just use `__class__` :)